controllers/argocd/util.go

```diff
@@ -348,7 +348,7 @@ func (r *ReconcileArgoCD) getArgoServerURI(cr *argoproj.ArgoCD) string {
 	}
 
 	// Use Route host if available, override Ingress if both exist
-	if IsRouteAPIAvailable() {
+	if cr.Spec.Server.Ingress.Enabled || cr.Spec.Server.Route.Enabled {
```

Comment:
Do you think we can remove the ingress check to avoid extra route check k8s call when route is not enabled?

Reply:
Ok. I will remove the extra check. If ingress is enabled in OpenShift, a route is created with a random name. That is the reason why I had it in the check. But since the random name won't match anyway, it does not make sense to have it in this check. So removing it.